Feature: board api #42

bjho606 opened the pull request:
close #42

chwangmin committed:
test1

bjho606 committed:
feat: 게시판 기본 CRUD 기능 구현 #42

- 게시판 생성, 숨기기, 삭제 기능 구현
- BoardService에 대한 테스트코드 작성 (Mock으로)

<할 일>
- [] response 형식 만들어서 반환하도록 수정
- [] BoardController에 대한 테스트코드 작성 (DB와 연동하여 테스트 진행)

bjho606 committed:
feat: 게시판 crud controller 까지 기능 구현 #42

- 게시판 생성, 숨기기, 삭제, 모든 게시판 불러오기 기능을 controller 단까지 구현

<할 일>
- controller 테스트코드 작성
- validation들 추가

bjho606 committed:
feat: 게시판 controller 테스트코드 작성 (불완전) #42

- Mock으로 테스트코드를 짜려고 했는데, 테스트는 동작하지만 실제 db에도 변화가 생기는 문제가 있음
- 테스트코드를 제대로 짠건지도 불확실함

bjho606 committed:
feat: 게시판 validation 추가 #42

chwangmin commented on src/main/java/com/ssafy/naem/config/secret/Secret.java:
민감한 정보 삭제해주세요 ~

chwangmin commented on src/main/java/com/ssafy/naem/domain/board/repository/BoardRepository.java:
필요없는 주석 삭제해주세요~

chwangmin reviewed (commented):
수정부탁드립니다~

chwangmin reviewed (approved):
board api 테스트 코드까지 짜시느라 고생많으셨습니다!!

minju412 commented on src/main/java/com/ssafy/naem/domain/board/controller/BoardController.java:
```diff
@@ -0,0 +1,94 @@
+    @GetMapping("/active")
+    public BaseResponse<BoardsResponse> getAllActiveBoards() {
+        try {
+            return new BaseResponse<>(boardService.getAllActiveBoards());
```
컨트롤러 단에서 모든 로직에 try~catch를 하고 있는데, 조금 더 고민해봅시다..!!

bjho606 replied:
ControllerAdvice로 처리 완료!

minju412 commented on src/main/java/com/ssafy/naem/domain/board/entity/Board.java:
```diff
@@ -0,0 +1,64 @@
+    @Builder.Default
+    private Status status = Status.STATUS_ACTIVE;
+
+    @PrePersist
```
제 코드에서 이 부분(생성, 수정일 갱신)을 BaseTimeEntity로 따로 정의해뒀으니 머지하고 수정하면 좋을 것 같아요!

minju412 reviewed (changes requested):
테스트코드까지 대단하네요👍 수정 후 다시 봅시다~!

bjho606 committed:
refactor: 게시판 Controller, Service, Response 리팩토링 #42

- try-catch 문을 ControllerAdvice로 공통 처리 구현
- BaseResponse에서 ErrorResponse 분리
- Controller와 Service 코드 정리

minju412 commented on src/main/java/com/ssafy/naem/domain/board/service/BoardService.java:
```diff
@@ -20,103 +20,79 @@
-        if (board.isPresent()) {
-            throw new BaseException(BaseResponseStatus.BOARD_ALREADY_EXISTS);
-        }
+        String newName = boardChangeNameRequest.name();
```
recored 문법 신기하네요👍

minju412 reviewed (approved):
Controller Advice까지 적용하느라 고생하셨습니다 👍👍

chwangmin reviewed (approved):
고생하셨습니다~
